mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -1300,6 +1366,33 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
     }
 
     const rejectAICode = (): void => {
+        const activeCellId = notebookTracker.activeCell?.model.id;
+        
+        // Agent review mode: reject specific cell from multi-cell review
+        if (activeCellId && cellStatesBeforeDiff.current.has(activeCellId)) {
+            const originalCode = cellStatesBeforeDiff.current.get(activeCellId);
+            if (originalCode === undefined) return;
+            
+            // Remove from tracking and restore original code
+            cellStatesBeforeDiff.current.delete(activeCellId);
+            writeCodeToCellByID(notebookTracker, originalCode, activeCellId);
+            turnOffDiffsForCell(notebookTracker, activeCellId, codeDiffStripesCompartments.current);
+            updateCellToolbarButtons();
+
+            // Re-run cell in background
+            void runCellByIDInBackground(notebookTracker.currentWidget, activeCellId);
```

Comment:
Oh interesting. It's not clear to me if we should or shouldn't be rerunning the cells when the user rejects the AI code. I guess you're right that rerunning makes sense. Otherwise the user is going to have to rerun the cells so that the agent has the correct version of all of the variables. Can we document that here?